src/main/java/com/seatapp/services/SeatService.java

```diff
@@ -79,35 +91,62 @@ private Seat getSeatById(final Long seatId) {
 
     /**
      * Reserves the seat with the specified id.
+     *
      * @param seatId         is the id of the to be reserved seat.
      * @param reservationDto is the reservation details.
+     * @param user           the user wanting to make a reservation.
      * @return the reserved seat.
      */
     public Seat reserve(final Long seatId,
-                        final ReservationDto reservationDto) {
+                        final ReservationDto reservationDto, final User user) {
         if (reservationDto == null) {
             throw new IllegalArgumentException("ReservationDto cannot be null");
         }
         Seat seat = getSeatById(seatId);
         Reservation newReservation = new Reservation(
-                reservationDto.getStartTime(), reservationDto.getEndTime());
+                reservationDto.getStartTime(),
+                reservationDto.getEndTime(), user);
         seat.addReservation(newReservation);
         seatRepository.save(seat);
         return seat;
     }
 
     /**
      * Gets all the seats with their reservations from the given date.
+     *
      * @param date is the date of the wanted reservations.
      * @return the list of seats.
      */
     public List<Seat> getAllWithReservationsByDate(final LocalDate date) {
         List<Seat> foundSeats = getAll();
-        for (Seat s :foundSeats) {
+        for (Seat s : foundSeats) {
             s.setReservations(s.getReservations().stream()
                     .filter(reservation -> reservation.getDate().equals(date))
                     .collect(Collectors.toList()));
         }
         return foundSeats;
     }
+
+    /**
+     * Checks in on the reservation of the seat.
+     *
+     * @param seatId   the seatId from the seat where you check in.
+     * @param username username of the person wanting to check in.
+     * @return the checked in seat.
+     */
+    public Seat checkInOnSeat(final Long seatId, final String username) {
+        Seat seat = getSeatById(seatId);
+
+        Reservation reservation = seat.getReservations().stream()
```

Comment:
Split this up in 2 filters and put `res.getStartTime().minusMinutes(Integer.parseInt(timeBeforeReservation))` in a seperate method to improve readability.